### DISCUSSION TITLE
Server-side vs client-side authentication with Next-auth (caching)

### DISCUSSION THREAD

**Discussion by codinginflow** [vercel/next.js#63858] [App Router]
I need some clarification here.

Let's say I want to display the currently logged-in user in the navbar.

![image](https://github.com/vercel/next.js/assets/52977034/89041072-9196-4b3b-bca2-7ac5133d6182)

If I fetch the session server-side (`getServerSession` or now just `await auth()` in v5), **all pages** that display this navbar are **dynamically rendered**. There is no way to **statically cache** them anymore, correct?

Considering this, would it make sense to move the session retrieval **client-side**, if I want to enable static caching? This would be done by wrapping the layout into a `SessionProvider`, **not** passing a `session` prop, and then retrieving the session via `useSession` inside the navbar.

    <html lang="en">
      <body className={inter.className}>
        // Not passing the session prop here will make the provider fetch it     
    client-side
        <SessionProvider>
          <Navbar />
          {children}
        </SessionProvider>
      </body>
    </html>

Am I missing something? Like another better option? Or any caveats about my approach?

**Comment by biryusal** [comment]
I guess so, because, `getServerSession` must be using `cookies()` from `nextjs`, and it makes the function become `dynamic`:
```
cookies() is a [Dynamic Function](https://nextjs.org/docs/app/building-your-application/rendering/server-components#server-rendering-strategies#dynamic-functions) whose returned values cannot be known ahead of time. Using it in a layout or page will opt a route into [dynamic rendering](https://nextjs.org/docs/app/building-your-application/rendering/server-components#dynamic-rendering) at request time.
```
https://nextjs.org/docs/app/api-reference/functions/cookies

`Considering this, would it make sense to move the session retrieval client-side, if I want to enable static caching?`
Yeah, it makes sense in my opinion

**Reply by codinginflow** [reply]
Thank you for the clarification!

**Reply by biryusal** [reply]
I am glad to help!
